subsys/bluetooth/host/gatt.c

```diff
@@ -1087,12 +1216,48 @@ static u8_t gatt_foreach_iter(const struct bt_gatt_attr *attr,
 	return result;
 }
 
+static void foreach_attr_type_dyndb(u16_t start_handle, u16_t end_handle,
+				    const struct bt_uuid *uuid,
+				    const void *attr_data, uint16_t num_matches,
+				    bt_gatt_attr_func_t func, void *user_data)
+{
+#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
+	int i;
+	struct bt_gatt_service *svc;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&db, svc, node) {
+		struct bt_gatt_service *next;
+
+		next = SYS_SLIST_PEEK_NEXT_CONTAINER(svc, node);
+		if (next) {
+			/* Skip ahead if start is not within service handles */
+			if (next->attrs[0].handle <= start_handle) {
+				continue;
+			}
+		}
+
+		for (i = 0; i < svc->attr_count; i++) {
+			struct bt_gatt_attr *attr = &svc->attrs[i];
+
+			if (gatt_foreach_iter(attr,
+						start_handle,
+						end_handle,
+						uuid, attr_data,
+						&num_matches,
+						func, user_data) ==
+				BT_GATT_ITER_STOP) {
+				return;
```

Comment:
Something is not right with the indentation here. BT_GATT_ITER_STOP shouldn't have the same indentation as the return. The split parameters in the gatt_foreach_iter call also seem to be wrong. You should be using tabs + 0-7 spaces to align with the right opening parenthesis of the first line.

Reply:
Fixed in latest push.